packages/flutter/lib/src/material/button_theme.dart

```diff
@@ -491,7 +491,7 @@ class ButtonThemeData extends Diagnosticable {
     if (fillColor != null)
       return fillColor;
 
-    if (button is FlatButton || button is OutlineButton)
+    if (button is FlatButton || button is OutlineButton || button.runtimeType == MaterialButton)
```

Comment:
Why "is" on the others, and runtimeType on the MaterialButton? Seems like they should all be one or the other, no?

Reply:
OK, maybe a comment explaining that would be good?